File: camp/week1/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int zero=0;
        int product=1;
        vector<int> answer;
        for(int i = 0 ; i<nums.size() ; i++){
            if(nums[i]==0)
                zero++;
            else{
                product*=nums[i];
            }
        }
        if(zero!=0){
            if(zero==1){
                for(int i=0;i<nums.size();i++){
                    if(nums[i]==0)
                        answer.push_back(product);
                    else
                        answer.push_back(0);
                }
            }
            else{
                for(int i=0;i<nums.size();i++){
                    answer.push_back(0);
                }
            }
        }
        else{
            for(int i=0;i<nums.size();i++){
                answer.push_back(product/nums[i]);
            }
        }
        return answer;
    }
};

File: camp/week1/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        pro = 1
        zero = 0
        for i in nums:
            if i == 0:
                zero += 1
            else:
                pro *= i
        for i in range(len(nums)):
            if zero > 1 or (nums[i] != 0 and zero > 0):
                nums[i] = 0
            elif nums[i] == 0 and zero == 1:
                nums[i] = pro
            else:
                nums[i] = pro // nums[i]
        return nums
